src/run.js

```diff
@@ -50,27 +50,39 @@ const getSha = (eventName, payload) => {
   return null
 }
 
+const getBranchFromGithubRef = (ref) => {
+  if (ref.startsWith('refs/')) {
+    const tokens = ref.split('/')
+    return (tokens.length === 3 ? tokens[2] : ref)
+  }
+  return ref
+}
```

Comment:
If we're simply returning the original ref if it's not the right format to extract a branch, logging an error message that says explains *why* the original `ref` is being returned will prevent some debugging headaches down the road.

Reply:
Done